litellm/redis_gpt_cache.py

```diff
@@ -0,0 +1,274 @@
+                            "ttl":int(os.getenv("CACHE_TTL",3600)) if os.getenv("CACHE_TTL",3600) else None
+                        },
+        "score_threshold":float(os.getenv("CACHE_SCORE_THRESHOLD",0.6)),
+        "metric_config":{
+                        "endpoint_id":uuid.uuid4(),
+                        "project_id":uuid.uuid4(),
+                        "model_id":uuid.uuid4(),
+                        "api_endpoint":"https://api.example.com/endpoint",
+                        "metric_request_id":uuid.uuid4(),
+                        "engine":"openai",
+                        "request_start_time":time.time(),
+                        "cache_metrics_enabled" : True
+                    }
+    }
+class BudServeMemoryCacheEviction(MemoryCacheEviction):
+    """ This class `BudServeMemoryCacheEviction` is a subclass of `MemoryCacheEviction`
+    that implements memory cache eviction policies such as LRU and TTL with customizable
+    parameters. """
+
+    def __init__(
+        self,
+        policy: str = "LRU",
+        maxsize: int = 1000,
+        clean_size: int = 0,
+        on_evict: Callable[[List[Any]], None] = None,
+        ttl: Optional[int] = None,
+        **kwargs,
+    ):
+        try:
+            super().__init__(policy, maxsize, clean_size, on_evict, **kwargs)
+        except ValueError:
+            self._policy = policy.upper()
+            if self._policy == "TTL":
+                if not ttl:
+                    raise ValueError("TTL policy requires ttl parameter")
+                self._cache = cachetools.TTLCache(maxsize=maxsize, ttl=ttl, **kwargs)
+            else:
+                raise ValueError(f"Unknown policy {policy}")
+            self._cache.popitem = popitem_wrapper(self._cache.popitem, on_evict, clean_size)
+
+def init_gptcache_redis(cache_obj: Cache, hashed_llm: str, cache_config: dict, embedding_model: str, similarity_threshold: float):
+    """Initialise the GPT cache object."""
+    print_verbose(f"Initialise the GPT cache object: init_gptcache_redis.{cache_config}")
+    endpoint_id = cache_config.get("endpoint_id", "1234")
+    embedding_model: str = cache_config.get("embedding_model", embedding_model)
+    try:
+        embeddings = HuggingFaceEmbeddings(
+            model_name=embedding_model,
+        )
+        embeddings = LangChain(embeddings=embeddings)
+    except Exception as exc:
+        print_verbose(f"gptcache redis semantic-cache embeddings error: {str(exc)}")
+        print_verbose(f"gptcache redis semantic-cache using Onnx embeddings")
+        embeddings = Onnx()
+    eviction_policy = cache_config.get("eviction_policy", {})
+
+    data_manager = manager_factory(
+        "redis,redis",
+        scalar_params={
+            "redis_host": os.getenv("REDIS_HOST", "localhost"),
+            "redis_port": os.getenv("REDIS_PORT", "6342"),
+            "password": os.getenv("REDIS_PASSWORD", "budpassword"),
```

Comment:
It's not alright to keep password values here. Please check how redis connection params are being set in RedisSemanticCache. I had kept it only for testing, but when we set up proxy server using config file, these values should be passed on from there. Please correct that.